src/libraries/System.IO.Packaging/src/System/IO/Packaging/ZipPackage.cs

```diff
@@ -383,7 +383,11 @@ internal static string GetOpcNameFromZipItemName(string zipItemName)
                     break;
                 case CompressionOption.Maximum:
                     {
+#if (!NETSTANDARD2_0 && !NETFRAMEWORK)
+                        compressionLevel = CompressionLevel.SmallestSize;
+#else
                         compressionLevel = CompressionLevel.Optimal;
+#endif
```

Comment:
I don't think conditioning to `NETSTANDARD2_0` would work correctly here. If the underlying framework being used is .NET Core/.NET, we would be going through the .NET Framework code. I think we should invert the #if and #else to:

```suggestion
#if (NETCOREAPP || NET)
                        compressionLevel = CompressionLevel.Optimal;
#else
                        
                        compressionLevel = CompressionLevel.SmallestSize;
#endif
```

cc @ViktorHofer for a quick confirmation.

Reply:
I originally did this to avoid changes to the CompressionLevel mapping for apps/libraries referencing the .NET Standard build of the package. I assumed that a .NET Core app would reference the .NET Core package and that only a .NET Framework app would need to reference .NET Standard, but I suppose that's not a guarantee.

I'm happy enough to use the `(NETCOREAPP || NET)` condition to make sure the behavioural change only appears in the .NET Core package, but I think it's backwards: we want compressionLevel to be SmallestSize for the .NET Core packages and Optimal for everything else.

Reply:
Precompile directives are resolved at compile and not run time. System.IO.Packaging targets `net9.0`, `net8.0`, `netstandard2.0` and `net462`.

> I'm happy enough to use the (NETCOREAPP || NET) condition to make sure the behavioural change only appears in the .NET Core package, but I think it's backwards: we want compressionLevel to be SmallestSize for the .NET Core packages and Optimal for everything else.

Sounds like you want the following?

```suggestion
#if NET
                        compressionLevel = CompressionLevel.SmallestSize;
#else
                        compressionLevel = CompressionLevel.Optimal;
#endif
```

Reply:
That's much neater, thanks - I'll use that instead

Reply:
@ViktorHofer yes, my bad. I inverted the code first and then decided to invert the ifs. Thank you for catching it!